Utilities/DataLoading/EdFi.LoadTools/ApiClient/FileContextProvider.cs

```diff
@@ -0,0 +1,152 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Schema;
+using EdFi.LoadTools.Engine;
+using log4net;
+
+namespace EdFi.LoadTools.ApiClient
+{
+    public class FileContextProvider : IFileContextProvider
+    {
+        private readonly IDataConfiguration _dataConfiguration;
+        private readonly Regex _interchangeNameRegex = new Regex(Constants.InterchangeNameRegex, RegexOptions.Compiled);
+        private readonly ILog _log = LogManager.GetLogger(nameof(FileContextProvider));
+        private readonly XmlSchemaSet _xmlSchemaSet;
+        private readonly IXsdConfiguration _xsdConfiguration;
+        private readonly Lazy<List<FileContext>> _fileContexts;
+
+        public FileContextProvider(XmlSchemaSet xmlSchemaSet, IDataConfiguration dataConfiguration,
+            IXsdConfiguration xsdConfiguration)
+        {
+            _xmlSchemaSet = xmlSchemaSet;
+            _dataConfiguration = dataConfiguration;
+            _xsdConfiguration = xsdConfiguration;
+
+            _fileContexts = new Lazy<List<FileContext>> (() => CreateFileContexts().Where(x => x.IsValid).ToList());
+        }
+
+        public IEnumerable<FileContext> GetFileContexts() => _fileContexts.Value;
+
+        private IEnumerable<FileContext> CreateFileContexts()
+        {
+            var fileContexts = new HashSet<FileContext>();
+
+            foreach (string file in Directory.GetFiles(_dataConfiguration.Folder, "*.xml", SearchOption.AllDirectories))
```

Comment:
This looks way cleaner, but this should also be responsible for generating a hashlist of resources it finds in the file, to be used when actually processing the files later.

Reply:
done